pyccel/codegen/printing/ccode.py

```diff
@@ -858,7 +934,30 @@ def _print_Assign(self, expr):
             self._temporary_args = [VariableAddress(a) for a in expr.lhs]
             return '{};'.format(self._print(expr.rhs))
         lhs = self._print(expr.lhs)
-        rhs = self._print(expr.rhs)
+        rhs = expr.rhs
+        if isinstance(rhs, (NumpyArray)):
+            if rhs.rank == 0:
+                raise NotImplementedError(expr.lhs + "=" + expr.rhs)
+            dummy_array_name, _ = create_incremented_string(self._parser.used_names, prefix = 'array_dummy')
+            dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
```

Comment:
Why don't you need to pass `array = True`?

Reply:
here i need the C type to use it in the declaration

Reply:
Hmm, maybe we need 2 `find_in_dtype_registry` functions. Using the same name gives me the impression that they are getting an object for the same role, so I would assume that you pass `array = Ture` when you are handling an array.

In reality `array = True` indicates that you want the name of the array data member, not the dtype for the declaration

Reply:
i will create another function to make it more clear